xcode/Subconscious/Shared/Components/Common/SubtextTextViewRepresentable.swift

```diff
@@ -402,10 +405,30 @@ struct SubtextTextViewRepresentable: UIViewRepresentable {
             context.coordinator.isUIViewUpdating = false
         }
 
-        if (view.text != state.text) {
+        // When a key is pressed, updateUIView is called multiple times. The first time, state has
+        // not yet been updated to hold the latest text BUT view.text has already updated.
+        // This can result in thrashing the TextView and killing performance.
+        
+        // Pathological scenario:
+        // state.text = "abc"
+        // view.text = "abc"
+        // > user types "d"
+        // view.text = "abcd"
+        // updateUIView() called
+        // we set view.text = state.text
+        // now, view.text = "abc"
+        // THEN state.text updates to "abcd"
+        // updateUIVIew() called again
+        // we set view.text = state.text
+        // view.text == "abcd"
+        
+        // so we set view.text TWICE when we could set it zero times
+        // caching the last known rendered value allows us to detect and sidestep this scenario
```

Comment:
In debugging I can see that action for `.setText` is actually dispatched before `updateUIView()` is called BUT the value stored in `state` lags behind the value passed to `.setText`. 

It's possible there's a solution to this problem at a layer below?